File: extract_all_tar.py
import glob
import os;
import  itertools;


for file in  itertools.chain(   glob.glob("*.tar.gz"), \
                                glob.glob("*.tgz")  \
                            ):
    print "echo tar -xzf  %s"%file;
    print "tar -xzf  %s"%file;


for file in  itertools.chain(   glob.glob("*.tar.bz2"), \
                                glob.glob("*.tbz"), \
                                glob.glob("*.tar.bz") \
                            ):
    print "echo tar -xjf  %s"%file;
    print "tar -xjf  %s"%file;

File: extract_all_tar.pl
#!/usr/bin/perl
opendir(DIR,".") or die("opendir error\n");
@files=readdir(DIR);

@file=grep { -f  && !/^\./ && /tar.gz$/} @files; 
for (@file)
{
    my $chm=$_;
    $chm=~s/\.tar.gz$/\.chm/g;
    if(! -f $chm)
    {
	    print "echo tar -xzf $_ \n";
	    print "tar -xzf $_ \n";
    }
}

@file=grep { -f  && !/^\./ && /tar$/} @files; 
for (@file)
{
	print "echo tar -xf $_ \n";
	print "tar -xf $_ \n";
}

@file=grep { -f  && !/^\./ && /tar.bz2$/} @files; 
for (@file)
{
	print "echo tar -xjf $_ \n";
	print "tar -xjf $_ \n";
}
closedir(DIR);



